cylc/flow/config.py

```diff
@@ -309,6 +310,15 @@ def __init__(
         self.cfg = self.pcfg.get(sparse=False)
         self.mem_log("config.py: after get(sparse=False)")
 
+        if self.cfg['cylc'].get('cycle point time zone') is None:
+            # Get the original suite run time zone if restart.
+            # This must be done before call to init_cyclers(self.cfg)
+            cp_tz_str = getattr(self.options, 'cp_tz', None)
+            if cp_tz_str is None:
+                # Not a restart
+                cp_tz_str = get_local_time_zone_format()
+            self.cfg['cylc']['cycle point time zone'] = cp_tz_str
```

Comment:
On the subject of "what if the user defines their own time zone", I think it's probably good enough to just issue a warning if:

* The time zone is set in the config
* And the time zone is changed by this logic.